core/src/main/java/org/elasticsearch/common/settings/SecureString.java

```diff
@@ -106,8 +106,29 @@ public synchronized String toString() {
      */
     @Override
     public synchronized void close() {
-        Arrays.fill(chars, '\0');
-        chars = null;
+        if (chars != null) {
+            Arrays.fill(chars, '\0');
+            chars = null;
+        }
+    }
+
+    /**
+     * Returns a closeable object that provides access to an array of chars that is a copy of this SecureString's backing chars. Upon
+     * closing this object the char array will be cleared without affecting this SecureString. This is useful for APIs which accept a
+     * char[]. For example:
+     *
+     * <pre>
+     *     try (CloseableChars closeableChars = secureString.getAsCloseableChars()) {
```

Comment:
I'm confused on why you wouldn't just close the secureString?  If we must have the ability get the the char[] so be it, and we add it to SecureString with sufficient warnings in the javadocs, but it seems this is adding an extra layer of indirection that would provide confusion, not utility.